src/ucs/type/spinlock.h

```diff
@@ -24,9 +24,24 @@ typedef struct ucs_spinlock {
 } ucs_spinlock_t;
 
 
-ucs_status_t ucs_spinlock_init(ucs_spinlock_t *lock);
+static inline ucs_status_t ucs_spinlock_init(ucs_spinlock_t *lock)
+{
+    int ret;
+
+    ret = pthread_spin_init(&lock->lock, 0);
+    if (ret != 0) {
+        return UCS_ERR_IO_ERROR;
+    }
 
-void ucs_spinlock_destroy(ucs_spinlock_t *lock);
+    lock->count = 0;
+    lock->owner = 0xfffffffful;
+    return UCS_OK;
+}
+
+static inline void ucs_spinlock_destroy(ucs_spinlock_t *lock)
```

Comment:
let's make `ucs_spinlock_destroy` returning `ucs_status_t`
```
static inline ucs_status_t ucs_spinlock_destroy(ucs_spinlock_t *lock)
{
    int ret;

    if (lock->count != 0) {
        return UCS_ERR_BUSY;
    }

    ret = pthread_spin_destroy(&lock->lock);
    if (ret != 0) {
        if (errno == EBUSY) {
            return UCS_ERR_BUSY;
        } else {
            return UCS_ERR_INVALID_PARAM;
        }
    }

    return UCS_OK;
}
```

the codes that call `ucs_spinlock_destroy` need to be updated to handle a return value from it

@yosefe wdyt?

Reply:
sounds good

Reply:
I'll change caller side of `ucs_spinlock_destroy` in _UCS_ module.

```C
status = ucs_spinlock_destroy(...)
if ( status != UCS_OK ) {
    ucs_warn("ucs_spinlock_destroy() failed (%d)", status);
}
```

Use `ucm_warn` instead of `ucs_warn` in UCM module.

```C
status = ucs_spinlock_destroy(...)
if ( status != UCS_OK ) {
    uct_warn("ucs_spinlock_destroy() failed (%d)", status);
}
```

Reply:
yes, here we should use `ucm_warn()`
in other places (where we can do it) - `ucs_warn()`